### ISSUE TITLE
block-scoped-var and function declaration hoisting

### ISSUE THREAD

**Issue by BerkeleyTrue** [eslint/eslint#2291]
Not sure if this is the intention of block-scoped-var or a bug.

I follow this pattern in most of my scripts

```js
module.exports = {
  foo: foo,
  bar: bar
};

function foo() {
}

function bar() {
}

```

This has started throwing errors in eslint with block-scoped-var 

Is this supposed to violate block-scoped-var?

using eslint 0.19.0

**Comment by nzakas** [comment]
What's the specific error message you're seeing? And is this new in 0.19.0, or did it also happen in 0.18.0?

**Comment by BerkeleyTrue** [comment]
`"foo" used outside of binding context       block-scoped-var`

This happens on both 0.18.0 and 0.19.0 with
`block-scoped-var: 2`

**Comment by nzakas** [comment]
Took a look, the rule is correct. With block scoping, a variable cannot be used before it is defined.

**Comment by BerkeleyTrue** [comment]
Thanks
